import_export/resources.py

```diff
@@ -1370,20 +1370,134 @@ def get_display_name(cls):
         return cls.__name__
 
 
-def modelresource_factory(model, resource_class=ModelResource, meta_options=None):
+def modelresource_factory(
+    model,
+    resource_class=ModelResource,
+    meta_options=None,
+    custom_fields=None,
+    dehydrate_methods=None,
+    hydrate_methods=None,
+):
     """
     Factory for creating ``ModelResource`` class for given Django model.
+    This factory function creates a ``ModelResource`` class dynamically, with support
+    for custom fields, methods.
+
+    :param model: Django model class
+
+    :param resource_class: Base resource class (default: ModelResource)
+
+    :param meta_options: Meta options dictionary
+
+    :param custom_fields: Dictionary mapping field names to Field object
+
+    :param dehydrate_methods: Dictionary mapping field names
+                              to dehydrate method (Callable)
+
+    :param hydrate_methods: Dictionary mapping field names to hydrate method (Callable)
+
+    ** Basic Usage: **
+
+    .. code-block:: python
+
+        # Simple resource creation
+        BookResource = modelresource_factory(model=Book)
+
+        # With meta options
+        BookResource = modelresource_factory(
+            model=Book,
+            meta_options={'fields': ('id', 'name'), 'use_bulk': True}
+        )
+
+    ** Usage:**
+
+    .. code-block:: python
+
+        from import_export.fields import Field
+        from import_export.widgets import CharWidget
+
+        # Complex example with multiple features
+        UserResource = modelresource_factory(
+            model=User,
+            meta_options={
+                'fields': ('id', 'username', 'email', 'full_name'),
+                'exclude': ('password',),
+                'widgets': {'email': {'coerce_to_string': True}}
+            },
+            custom_fields={
+                'full_name': Field(
+                    attribute='get_full_name',
+                    column_name='Full Name',
+                    readonly=True
+                )
+            },
+            dehydrate_methods={
+                'full_name': lambda obj: f"{obj.first_name} {obj.last_name}".title()
+            }
+        )
+
+    :returns: ModelResource class
     """
+
+    def _create_dehydrate_func_wrapper(func):
+        def wrapper(self, obj):
+            return func(obj)
+
+        return wrapper
+
+    def _create_hydrate_func_wrapper(func):
```

Comment:
I notice that you support `hydrate_methods` - however we have no concept of this in the existing codebase, so this would be an undocumented feature.  I would suggest to remove this because we have other documented ways of modifying values on import.